src/main/java/fruitshop/service/handler/SupplyOperation.java

```
@@ -0,0 +1,16 @@
+package fruitshop.service.handler;
+
+import fruitshop.service.common.FruitTransaction;
+import fruitshop.storage.Storage;
+
+public class SupplyOperation implements OperationHandler {
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getQuantity() > 0) {
+            Storage.add(transaction.getFruit(), transaction.getQuantity());
+        } else {
+            System.out.println("Invalid quantity. Can not supply 0 product.");
```

This violates checklist item: 'Remember... quantity less than zero' and 'Don't ignore exceptions in try-catch blocks'. If the quantity is not positive, you must throw a RuntimeException, not just print a message. Also, you should validate that the fruit name is not null or empty.